sqlglot/generator.py

```diff
@@ -2063,7 +2070,19 @@ def select_sql(self, expression: exp.Select) -> str:
             self.sql(expression, "into", comment=False),
             self.sql(expression, "from", comment=False),
         )
-        return self.prepend_ctes(expression, sql)
+
+        sql = self.prepend_ctes(expression, sql)
+
+        if self.SUPPORT_SELECT_INTO == False and into:
+            # Ensures that SELECT INTO TEMP UNLOGGED -> CREATE TEMPORARY TABLE
+            kind = (
+                " TEMPORARY"
+                if into.args.get("temporary")
+                else (" UNLOGGED" if into.args.get("unlogged") else "")
+            )
+            sql = f'CREATE{kind} TABLE {into.args["this"]} AS {sql}'
```

Comment:
- `.this` is a shorthand for `.args["this"]`
- you shouldn't interpolate expressions because they're implicitly generated using the default dialect, you need to make sure the generation goes through the `sql` method which carries on the dialect information

```suggestion
            sql = f'CREATE{kind} TABLE {self.sql(into.this)} AS {sql}'
```